src/fhirtypes/ElementDefinition.ts

```diff
@@ -498,8 +784,13 @@ export class ElementDefinition {
    *   be resolved.
    */
   unfold(resolve: ResolveFn = () => undefined): ElementDefinition[] {
-    if (this.type.length === 1) {
-      const def = resolve(this.type[0]);
+    if (
+      this.type.length === 1 &&
+      (this.type[0].profile == null || this.type[0].profile.length <= 1)
+    ) {
+      // If it has a profile, use that, otherwise use the code
+      const type = this.type[0].profile?.[0] ?? this.type[0].code;
```

Comment:
This is the most beautiful line of code I've ever seen.

Reply:
There are a lot of `?`s, but it's definitely better than:
```js
let type;
if (this.type[0].profile && this.type[0].profile.length > 0) {
  type = this.type[0].profile[0];
} else {
  type = this.type[0].code;
}
```